### PR TITLE
FreeBSD compatibility patches.

### PR DESCRIPTION
**Pull request by yurivict**

### CODE REVIEW

File: src/JITModule.cpp [hunk 1]
```diff
@@ -109,6 +109,17 @@ void load_opengl() {
         llvm::sys::DynamicLibrary::LoadLibraryPermanently("/System/Library/Frameworks/OpenGL.framework/OpenGL", &error);
         user_assert(error.empty()) << "Could not find OpenGL.framework\n";
     }
+#elif defined(__FreeBSD__)|| defined(__OpenBSD__) || defined( __NetBSD__) || defined(__DragonFly__)
```
Comment (abadams) [inline src/JITModule.cpp:112]: Looks like clang-format doesn't like the space before `__NetBSD__`

**Comment by steven-johnson** [comment]
A complete rewrite of the CMake build system will be landing soon (possibly later today), so the change to halide.cmake will need attention then

**Comment by abadams** [comment]
Thanks for doing this, looks like the changes are small. We don't really have any way of testing BSD though so I guess we'll have to rely on BSD users to fix things after the fact if something unique to BSD breaks.

**Comment by krytarowski** [comment]
Please *never* put all BSDs into a shared type. This code is invalid for !FreeBSD, at least for NetBSD. If you want to add FreeBSD support, please define it as FreeBSD only.

**Comment by krytarowski** [comment]
OpenBSD is also broken by this.

**Comment by krytarowski** [comment]
Also, nobody will ever test it on other BSD variants such as BSD/OS.

**Comment by abadams** [comment]
Seems like changing the enum and target flag to FreeBSD/freebsd would address that?

**Comment by abadams** [comment]
I guess also the ifdef

**Comment by krytarowski** [comment]
Please treat FreeBSD as a separate OS. They diverged 25 years ago!

**Comment by yurivict** [comment]
I've updated the pull request.

File: src/Util.cpp [hunk 2]
```diff
@@ -20,6 +20,12 @@
 #include <sys/stat.h>
 #include <sys/types.h>
 
+#if defined(__FreeBSD__)
+#define CAN_GET_RUNNING_PROGRAM_NAME
+#include <sys/types.h>
```
Comment (steven-johnson) [inline src/Util.cpp:25]: To satisfy our clang-format rules, these must be alphabetically sorted:

```
#include <sys/sysctl.h>
#include <sys/syslimits.h>  // For PATH_MAX
#include <sys/types.h>
```

**Comment by steven-johnson** [comment]
Changes look good (pending buildbot testing in https://github.com/halide/Halide/pull/4962), with the caveat that we need to add information to the README (and to a few files, e.g. Target.h) indicating that FreeBSD isn't regularly tested by the core contributors, and so may have bugs, yadda yadda

**Comment by steven-johnson** [comment]
This needs attention as of now, the halide.cmake file no longer exists

**Comment by steven-johnson** [comment]
(Can you re-sync this with master and resolve the conflicts?)

**Comment by steven-johnson** [comment]
Reminder: we would welcome this patch if it's synced to master and updated appropriately. 

(If not, we may close it in a few days due to inattention; if that happens, feel free to reopen it in the future.)

**Comment by steven-johnson** [comment]
Closing due to inactivity. (Feel free to re-open.)

**Pull request closed without merging**